internal/flavors/assetinventory/strategy_aws.go

```diff
@@ -0,0 +1,113 @@
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package assetinventory
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/internal/config"
+	"github.com/elastic/cloudbeat/internal/inventory"
+	"github.com/elastic/cloudbeat/internal/inventory/awsfetcher"
+	"github.com/elastic/cloudbeat/internal/resources/providers/awslib"
+	"github.com/elastic/cloudbeat/internal/resources/utils/pointers"
+)
+
+const (
+	rootRole   = "cloudbeat-asset-inventory-root"
+	memberRole = "cloudbeat-asset-inventory-securityaudit"
+)
+
+func (s *strategy) initAwsFetchers(ctx context.Context) ([]inventory.AssetFetcher, error) {
+	// TODO(kuba): Handle AWS Gov
+	awsConfig, err := awslib.InitializeAWSConfig(s.cfg.CloudConfig.Aws.Cred)
+	if err != nil {
+		return nil, err
+	}
+
+	idProvider := awslib.IdentityProvider{Logger: s.logger}
+	awsIdentity, err := idProvider.GetIdentity(ctx, *awsConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Early exit if we're scanning the entire account.
+	if s.cfg.CloudConfig.Aws.AccountType == config.SingleAccount {
+		return awsfetcher.New(s.logger, awsIdentity, *awsConfig), nil
+	}
+	if s.cfg.CloudConfig.Aws.AccountType != config.OrganizationAccount {
+		return nil, fmt.Errorf("unsupported account_type: %q", s.cfg.CloudConfig.Aws.AccountType)
+	}
+
+	// Assume audit roles per selected account and generate fetchers for them
+	rootRoleConfig := assumeRole(
+		sts.NewFromConfig(*awsConfig),
+		*awsConfig,
+		fmtIAMRole(awsIdentity.Account, rootRole),
+	)
+	accountProvider := &awslib.AccountProvider{}
+	accountIdentities, err := accountProvider.ListAccounts(ctx, s.logger, rootRoleConfig)
+	if err != nil {
+		return nil, err
+	}
+	var fetchers []inventory.AssetFetcher
+	stsClient := sts.NewFromConfig(rootRoleConfig)
+	for _, identity := range accountIdentities {
+		assumedRoleConfig := assumeRole(
+			stsClient,
+			rootRoleConfig,
+			fmtIAMRole(identity.Account, memberRole),
+		)
+		if ok := tryListingBuckets(ctx, s.logger, assumedRoleConfig); !ok {
+			// role does not exist, skip identity/account
```

Comment:
Maybe a warn log so we have registered that certain accounts aren't scanned by the lack of a role with a specific name?

Reply:
I did not want to warn about something we consider normal behavior, but I can log info.

Reply:
Done